### DISCUSSION TITLE
Add cog update from private Github Repos

### DISCUSSION THREAD

**Discussion by entchen66** [kkrypt0nn/Python-Discord-Bot-Template#77] [Ideas]
Hey,

it would be cool to add a default repo cog, that will allow to update cogs from a dedicated repository. It would make developing cogs for the Tempalte more easy when the bot is running on a server.

Hint: a private rsa key ahve to be add to the server where the bot is located, so the git commands can access the cogs. Another way is to use the modern method with git access_tokens.

Greetings
entchen

**Comment by kkrypt0nn** [comment]
Hey there,

That will not be added for obvious reasons: it's a *template to get started into bot development with Python*, not a fully fledged bot. You can do that for your own bot - based off this template or not.

The goal of this template is **not** for people to download and run the bot and have everything done, it's supposed to be a starting point for new people to get an idea around what can be done and how it should be done.

Same reason why I will not be adding Docker related files.

Best regards
Krypton

**Reply by entchen66** [reply]
Thank you for your response, I understand your point that the template is meant to be a starting point for new developers to learn from. I can see how it can serve as a great foundation for building a fully-fledged bot. I agree that it is important for the template to focus on providing the basics for learning, instead of including all the additional features that a fully-fledged bot might have. Thanks for your insight and providing such a great resource for learning.

Greetings
